jme3-effects/src/main/java/com/jme3/post/filters/ContrastAdjustmentFilter.java

```diff
@@ -0,0 +1,309 @@
+ *
+ * * Redistributions in binary form must reproduce the above copyright
+ *   notice, this list of conditions and the following disclaimer in the
+ *   documentation and/or other materials provided with the distribution.
+ *
+ * * Neither the name of 'jMonkeyEngine' nor the names of its contributors
+ *   may be used to endorse or promote products derived from this software
+ *   without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED
+ * TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+ * LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+ * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.jme3.post.filters;
+
+import com.jme3.asset.AssetManager;
+import com.jme3.export.InputCapsule;
+import com.jme3.export.JmeExporter;
+import com.jme3.export.JmeImporter;
+import com.jme3.export.OutputCapsule;
+import com.jme3.material.Material;
+import com.jme3.math.Vector2f;
+import com.jme3.math.Vector3f;
+import com.jme3.post.Filter;
+import com.jme3.renderer.RenderManager;
+import com.jme3.renderer.ViewPort;
+import java.io.IOException;
+
+/**
+ * A color filter used to change the contrast for each color channel independently and the brightness of colors
+ * of the scene textures based on a simple transfer function.
+ *
+ * The transfer function involves setting the brightness of colors based on a maxValue and a minValue,
+ * then setting the contrast of each color channel independently using a power law function before the textures' colors are processed by the
+ * rasterizer and the final result can be scaled again on runtime using scale factors from 0.0 to 1.0.
+ *
+ * @author pavl_g.
+ */
+public class ContrastAdjustmentFilter extends Filter {
+    //the different channels exponents
+    protected float exp_r = 2.2f;
+    protected float exp_g = 2.2f;
+    protected float exp_b = 2.2f;
+    //the minimum value and the maximum value of the brightness parameter
+    protected float minBrightness = 0f;
+    protected float maxBrightness = 1f;
+    //the final pass scale factor
+    protected float scale_r = 1f;
+    protected float scale_b = 1f;
+    protected float scale_g = 1f;
+    protected Material material;
+
+    /**
+     * Instantiates a default color contrast filter, default brightness and default scale.
+     * Default values :
+     * - Exponents = 2.2 on all channels.
```

Comment:
2.2 is a strange choice for a default. I think the default exponents should all be 1.

Reply:
That would leave the `default` constructor with no use.

If i am going to make it 1.0, i will remove the default constructor (the first constructor on this class)...so what do you think ?